ShipEngineSDK/Api/CarriersApi.cs

```diff
@@ -0,0 +1,631 @@
+/*
+ * ShipEngine API
+ *
+ * The version of the OpenAPI document: 1.1.202406212006
+ * Contact: sales@shipengine.com
+ * Generated by: https://github.com/openapitools/openapi-generator.git
+ */
+
+
+using ShipEngineSDK.Model;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Mime;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ShipEngineSDK;
+
+/// <summary>
+/// Represents a collection of functions to interact with the API endpoints
+/// </summary>
+public partial interface IShipEngine
+{
+    /// <summary>
+    /// Add Funds To Carrier Add Funds To A Carrier
+    /// </summary>
+    /// <exception cref="System.ArgumentNullException">Thrown when required argument is null</exception>
+    /// <exception cref="ShipEngineSDK.ShipEngineException">Thrown when fails to make API call</exception>
+    /// <param name="carrierId">Carrier ID</param>
+    /// <param name="addFundsToCarrierRequestBody"></param>
+    /// <param name="cancellationToken">Cancellation Token to cancel the request.</param>
+    /// <returns>Task of ApiResponse (AddFundsToCarrierResponseBody)</returns>
+    Task<AddFundsToCarrierResponseBody> AddFundsToCarrier(string carrierId, AddFundsToCarrierRequestBody addFundsToCarrierRequestBody, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Add Funds To Carrier Add Funds To A Carrier
+    /// </summary>
+    /// <exception cref="System.ArgumentNullException">Thrown when required argument is null</exception>
+    /// <exception cref="ShipEngineSDK.ShipEngineException">Thrown when fails to make API call</exception>
+    /// <param name="carrierId">Carrier ID</param>
+    /// <param name="addFundsToCarrierRequestBody"></param>
+    /// <param name="cancellationToken">Cancellation Token to cancel the request.</param>
+    /// <returns>Task of ApiResponse (AddFundsToCarrierResponseBody)</returns>
+    Task<AddFundsToCarrierResponseBody> AddFundsToCarrier(Config methodConfig, string carrierId, AddFundsToCarrierRequestBody addFundsToCarrierRequestBody, CancellationToken cancellationToken = default);
```

Comment:
If we are breaking compatibility anyway, I think we should get rid of the ability to pass a `Config` to methods.  It has a very limited arbitrary set of things it lets you change. There are better ways to modify the way the request is performed (like passing an `Action<HttpRequestMessage>` or similar.

Reply:
I'm happy to get rid of this overload.  I just added it to be as close to how the API currently works.

Reply:
RIght. It is something I wanted to get rid of, but would mean a big breaking change. But if we are making a breaking change anyway, we should strongly consider if it is the best way to convey that type of information. 
@jpill may have thoughts on the subject.

Reply:
I'm not married to keeping the `Config` approaches, I agree that there are now improved paths to accomplish what that does.